pymc/distributions/mixture.py

```diff
@@ -112,457 +125,249 @@ class Mixture(Distribution):
+                f"Mixture components must all have the same dimensionality, got {components_ndim}"
             )
-        shape = kwargs.pop("shape", ())
 
-        self.w = w = at.as_tensor_variable(w)
-        self.comp_dists = comp_dists
+        if len(components_ndim_supp) > 1:
+            raise ValueError(
+                f"Mixture components must all have the same support dimensionality, got {components_ndim_supp}"
+            )
 
-        defaults = kwargs.pop("defaults", [])
+        w = at.as_tensor_variable(w)
+        return super().dist([w, *comp_dists], **kwargs)
 
-        if all_discrete(comp_dists):
-            default_dtype = _conversion_map[aesara.config.floatX]
+    @classmethod
+    def rv_op(cls, weights, *components, size=None, rngs=None):
+        # Update rngs if provided
+        if rngs is not None:
+            components = cls._reseed_components(rngs, *components)
+            *_, mix_indexes_rng = rngs
         else:
-            default_dtype = aesara.config.floatX
-
-            try:
-                self.mean = (w * self._comp_means()).sum(axis=-1)
-
-                if "mean" not in defaults:
-                    defaults.append("mean")
-            except AttributeError:
-                pass
-        dtype = kwargs.pop("dtype", default_dtype)
-
-        try:
-            if isinstance(comp_dists, Distribution):
-                comp_mode_logps = comp_dists.logp(comp_dists.mode)
-            else:
-                comp_mode_logps = at.stack([cd.logp(cd.mode) for cd in comp_dists])
-
-            mode_idx = at.argmax(at.log(w) + comp_mode_logps, axis=-1)
-            self.mode = self._comp_modes()[mode_idx]
-
-            if "mode" not in defaults:
-                defaults.append("mode")
-        except (AttributeError, ValueError, IndexError):
-            pass
-
-        super().__init__(shape, dtype, defaults=defaults, *args, **kwargs)
-
-    @property
-    def comp_dists(self):
-        return self._comp_dists
-
-    @comp_dists.setter
-    def comp_dists(self, comp_dists):
-        self._comp_dists = comp_dists
-        if isinstance(comp_dists, Distribution):
-            self._comp_dist_shapes = to_tuple(comp_dists.shape)
-            self._broadcast_shape = self._comp_dist_shapes
-            self.comp_is_distribution = True
+            # Create new rng for the mix_indexes internal RV
+            mix_indexes_rng = aesara.shared(np.random.default_rng())
+
+        if size is not None:
+            components = cls._resize_components(size, *components)
+
+        single_component = len(components) == 1
+
+        # Extract support and replication ndims from components and weights
+        ndim_supp = components[0].owner.op.ndim_supp
+        ndim_batch = components[0].ndim - ndim_supp
+        if single_component:
+            # One dimension is taken by the mixture axis in the single component case
+            ndim_batch -= 1
+        weights_ndim_batch = max(0, weights.ndim - ndim_batch - 1)
```

Comment:
`weights_ndim_batch` is the extra dimensions besides what is implied by the components dimensions. It can be arbitrarily large.
```python
# In this example, weights_ndim_batch = 4
pm.Mixture(w=(np.full((4, 4, 4, 4, 4, 2), comp_dists=pm.Normal.dist(size=(4, 2)))`
```

Reply:
I added a comment to explain better in the code